redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+namespace redfish
+{
+
+/**
+ * @brief Get power supply average and date values given chassis and power
+ * supply IDs.
+ *
+ * @param[in] aResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisID - Chassis to which the values are associated.
+ * @param[in] powerSupplyID - Power supply to which the values are associated.
+ *
+ * @return None.
+ */
+inline void getAverageValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const std::string& chassisID,
+                             const std::string& powerSupplyID)
+{
+    BMCWEB_LOG_INFO << "Get power supply average input power values";
+    BMCWEB_LOG_DEBUG << "ChassisID: " << chassisID;
+    BMCWEB_LOG_DEBUG << "PowerSupplyID: " << powerSupplyID;
+    // Setup date/timestamp and average values as arrays.
+    aResp->res.jsonValue["Date"] = nlohmann::json::array();
```

Comment:
I thought these would be under Oem Ibm? E.g. "Oem""IBM""Date" ?

Reply:
I had code here where I tried to fill it in that way, it ended up commented out, and I removed it before creating this PR, as it appeared to be showing up where it should. Maybe I dreamed that up and it is not in the right spots.

Reply:
Do I need to click that "Start a review" button to get these replies to show up?